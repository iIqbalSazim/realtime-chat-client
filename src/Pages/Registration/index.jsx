import { Link } from "react-router-dom";
import RegistrationForm from "./Form/RegistrationForm";

const Registration = () => {
  return (
    <main className="h-screen flex-col w-screen flex justify-center items-center">
      <RegistrationForm />
      <div>
        Already have an account?{" "}
        <Link to={"/"} className="btn-link">
          Login
        </Link>
      </div>
    </main>
  );
};

export default Registration;
