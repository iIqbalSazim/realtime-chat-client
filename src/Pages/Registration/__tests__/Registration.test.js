import { screen } from "@testing-library/react";

import { renderWithProviders } from "../../../../jest.setup";

import Registration from "../Registration";

describe("Registration", () => {
  test("renders correctly", () => {
    const { container } = renderWithProviders(<Registration />);

    const alreadyHaveAnAccount = screen.getByText(/Already have an account?/i);
    const loginLink = screen.getByRole("link", { name: /login/i });

    expect(alreadyHaveAnAccount).toBeInTheDocument();
    expect(loginLink).toBeInTheDocument();
    expect(loginLink).toHaveAttribute("href", "/");

    expect(container).toMatchSnapshot();
  });
});
