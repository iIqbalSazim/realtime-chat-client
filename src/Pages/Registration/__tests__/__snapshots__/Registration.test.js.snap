// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration renders correctly 1`] = `
<div>
  <main
    class="h-screen flex-col w-screen flex justify-center items-center"
  >
    <form
      class="flex flex-col gap-4 items-center justify-center w-[500px] px-10 py-12 border border-slate-200"
    >
      <h1
        class="font-medium text-lg mb-5"
      >
        REGISTER
      </h1>
      <div
        class="form-control w-full max-w-xs"
      >
        <label
          class="label"
        >
          <span
            class="label-text"
          >
            Username
          </span>
        </label>
        <input
          class="input input-bordered w-full max-w-xs"
          name="username"
          placeholder="Your username"
          type="text"
        />
      </div>
      <div
        class="form-control w-full max-w-xs"
      >
        <label
          class="label"
        >
          <span
            class="label-text"
          >
            Password
          </span>
        </label>
        <input
          class="input input-bordered w-full max-w-xs"
          name="password"
          placeholder="Your password"
          type="password"
        />
      </div>
      <button
        class="btn btn-md btn-secondary font-semibold mt-5"
        type="submit"
      >
        Submit
      </button>
    </form>
    <div>
      Already have an account?
       
      <a
        class="btn-link"
        href="/"
      >
        Login
      </a>
    </div>
  </main>
</div>
`;
